src/CustomList.php

```diff
@@ -129,6 +157,25 @@ public function add(mixed $item): void
         $this->items[] = $item;
     }
 
+    /**
+     * Remove duplicates of list
+     *
+     * @return void
+     * @throws CustomListTypeErrorException
+     */
+    public function unique(): CustomList
+    {
+        $result = new static();
+
+        foreach ($this->items as $item) {
+            if(!$result->contains($item)) {
+                $result->add($item);
+            }
+        }
+
+        return $result;
+    }
```

Comment:
We could use the array functions to filter the duplicated values:

```suggestion
     public function unique(): CustomList
    {
        $this->items = array_values(array_unique($this->items, SORT_REGULAR));

        return $this;
    }
```

Using `SORT_REGULAR` the [array_unique](https://www.php.net/manual/en/function.array-unique.php#:~:text=SORT_REGULAR%20%2D%20compare%20items%20normally%20(don%27t%20change%20types)) compares using `==` instead `===` just like the function "contains" does so if we use this suggestion the contains function could be removed because we don't need it.